scripts/setup_ebpf_cicd_tests.ps1

```diff
@@ -50,16 +48,87 @@ if ($TestMode -eq "CI/CD" -or $TestMode -eq "Regression") {
 Get-CoreNetTools
 Get-PSExec
 
-# Export build artifacts to the test VMs.
-Export-BuildArtifactsToVMs -VMList $VMList -ErrorAction Stop
+$Job = Start-Job -ScriptBlock {
+    param ([Parameter(Mandatory = $True)] [PSCredential] $TestVMCredential,
+           [Parameter(Mandatory = $true)] [PSCustomObject] $Config,
+           [Parameter(Mandatory = $true)] [string] $SelfHostedRunnerName,
+           [parameter(Mandatory = $true)] [string] $TestMode,
+           [parameter(Mandatory = $true)] [string] $LogFileName,
+           [parameter(Mandatory = $true)] [string] $WorkingDirectory = $pwd.ToString(),
+           [parameter(Mandatory = $true)] [bool] $KmTracing,
+           [parameter(Mandatory = $true)] [string] $KmTraceType
+    )
+    Push-Location $WorkingDirectory
+
+    # Load other utility modules.
+    Import-Module .\common.psm1 -Force -ArgumentList ($LogFileName) -WarningAction SilentlyContinue
+    Import-Module .\config_test_vm.psm1 -Force -ArgumentList ($TestVMCredential.UserName, $TestVMCredential.Password, $WorkingDirectory, $LogFileName) -WarningAction SilentlyContinue
+
+    $VMList = $Config.VMMap.$SelfHostedRunnerName
+
+    # Get all VMs to ready state.
+    Initialize-AllVMs -VMList $VMList -ErrorAction Stop
+
+    # Export build artifacts to the test VMs.
+    Export-BuildArtifactsToVMs -VMList $VMList -ErrorAction Stop
+
+    # Configure network adapters on VMs.
+    Initialize-NetworkInterfacesOnVMs $VMList -ErrorAction Stop
+
+    # Install eBPF Components on the test VM.
+    foreach($VM in $VMList) {
+        $VMName = $VM.Name
+        Install-eBPFComponentsOnVM -VMName $VMname -TestMode $TestMode -KmTracing $KmTracing -KmTraceType $KmTraceType -ErrorAction Stop
+    }
+
+    Pop-Location
+}  -ArgumentList (
+    $TestVMCredential,
+    $TestExecutionConfig,
+    $SelfHostedRunnerName,
+    $TestMode,
+    $LogFileName,
+    $WorkingDirectory,
+    $KmTracing,
+    $KmTraceType)
+
+# Keep track of the last received output count
+$TimeElapsed = 0
+$JobTimedOut = $false
+
+# Loop to fetch and print job output in near real-time
+while ($Job.State -eq 'Running') {
+    $JobOutput = Receive-Job -Job $job
+	$JobOutput | ForEach-Object { Write-Host $_ }
+
+    Start-Sleep -Seconds 2
+    $TimeElapsed += 2
+
+    if ($TimeElapsed -gt $TestJobTimeout) {
```

Comment:
nit: this code is repeated in 3 different files. Is it possible to move this to a common function?